src/cephadm/cephadm.py

```diff
@@ -3070,6 +3074,18 @@ def get_container_mounts(ctx, fsid, daemon_type, daemon_id,
             data_dir = re.sub(r'\.tcmu$', '', data_dir)
         log_dir = get_log_dir(fsid, ctx.log_dir)
         mounts.update(CephIscsi.get_container_mounts(data_dir, log_dir))
+        # Modifications podman makes to /etc/hosts causes issues with
+        # iscsi (specifically referencing "host.containers.internal" entry
+        # being added to /etc/hosts in this case). To avoid that, but still
+        # allow users to use /etc/hosts for hostname resolution, we can
+        # mount a copy of the host's /etc/hosts file. The copy of the hosts file
+        # is made later in the "create_daemon_dirs" function of the CephIscsi class.
```

Comment:
Given https://tracker.ceph.com/issues/57018, I wonder if this should be made more general and not specific to iSCSI?